docroot/profiles/custom/cgov_site/modules/custom/cgov_infographic/cgov_infographic.routing.yml

```diff
@@ -0,0 +1,9 @@
+# Set up route for retrieving an infographic's long description text.
+cgov_infographic.long_description:
+  path: '/cgov_site/cgov_infographic/long-description/{node}'
```

Comment:
We discussed this following the Drupal pattern of `/{entity}/id/{function}`, as in `/media/{media}/long-description`.  Did you find out this would not work, or there were other issues?

No matter what, `{node}` would be inappropriate in this case anyway since an infographic is of type media. So `{node}` needs to be `{media}`.

It looks like if you add the following to the route (after `requirements:`, it can load the entity for you:
```
  options:
    parameters:
      media:
        type: entity:media
```
If you truly just want the `{media}` to represent the id, then you can also add a requirement to validate the ID of:
```
requirements:
   _permission: 'access content'
  media: \d+
```

Finally, and this can be a backlog ticket, we should see if there is a requirement that says, only do this if the media has the accessible text field.

Reply:
> we should see if there is a requirement that says, only do this if the media has the accessible text field

There was a Slack conversation with @lburack regarding required fields vs. governance for making sure there was either a body or an accessible text version.  Acting on the assumption that it wouldn't make sense to link to a blank long description, I coded the longdesc attribute to only be generated if accessible text existed.   (If you try to access a non-existant accessible text version, you'll get a 404.)